pkg/devfile/adapters/docker/component/utils.go

```diff
@@ -359,7 +361,7 @@ func (a Adapter) execDevfile(commandsMap common.PushCommandsMap, componentExists
 
 			containerID := utils.GetContainerIDForAlias(containers, command.Exec.Component)
 			compInfo := common.ComponentInfo{ContainerName: containerID}
-			err = exec.ExecuteDevfileBuildAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show)
+			err = exec.ExecuteDevfileBuildAction(&a.Client, *command.Exec, command.Exec.Id, compInfo, show, a.machineEventLogger)
 			if err != nil {
 				return err
```

Comment:
Should the machineEventLogger return an error here? Since the regular error output will be swallowed

Reply:
This is covered by the called `exec.ExecuteDevfile*` functions: in all cases the error will logged as part of the DevFileCommandExecutionComplete in these functions ([for example](https://github.com/jgwest/odo/blob/30058c589ec3faa80dbcb72b3bde5c58eb9a3268/pkg/exec/devfile.go#L47)). `ReportError` in the machine logger does exist for this purpose, but it would be a double-print of the same error in this case.